src/App.tsx

```diff
@@ -1,16 +1,202 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorNotification }
+  from './components/ErrorNotification/ErrorNotification';
+import {
+  addTodo, deleteTodo, editTodo, getTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { ErrorMessage } from './types/ErrorMessage';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [selectedTodoIds, setSelectedTodoIds] = useState<number[]>([]);
+
+  const activeTodosQuantity = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const isAnyTodoCompleted = useMemo(() => (
+    todos.some(todo => todo.completed)
+  ), [todos]);
+
+  const isEachTodoCompleted = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterStatus) {
+        case FilterStatus.Active:
+          return !todo.completed;
+
+        case FilterStatus.Completed:
+          return todo.completed;
+
+        case FilterStatus.All:
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterStatus]);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorMessage(ErrorMessage.UnableToLoad));
+    }
+  }, []);
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!title.trim()) {
+        setErrorMessage(ErrorMessage.TitleCantBeEmpty);
+
+        return;
+      }
+
+      const addNewTodo = async () => {
+        setIsAdding(true);
```

Comment:
it's good way to separate logic of adding todo and validation (above)
but would better to keep it outside handleSubmit and wrap with useCallback